cmd/finch/nerdctl.go

```diff
@@ -121,11 +124,13 @@ func (nc *nerdctlCommand) run(cmdName string, args []string) error {
 		case strings.HasPrefix(arg, "--add-host"):
 			switch arg {
 			case "--add-host":
-				args[i+1] = resolveIP(args[i+1], nc.logger)
+				args[i+1] = resolveIP(nc.fc, args[i+1], nc.logger)
 			default:
-				resolvedIP := resolveIP(arg[11:], nc.logger)
+				resolvedIP := resolveIP(nc.fc, arg[11:], nc.logger)
 				arg = fmt.Sprintf("%s%s", arg[0:11], resolvedIP)
 			}
+			nerdctlArgs = append(nerdctlArgs, "--host-gateway-ip")
```

Comment:
Why do we need to append this?

Reply:
This is for adding the host-gateway-ip into the nerdctl, other option might be adding it during the init

Reply:
The logic around here is to do special processing for specific args. If --host-gateway-ip can be directly passed to nerdctl and doesn't require special processing, I guess you don't need to change anything.

Reply:
@ningziwen Initially I plan to change the resolveIP function in the nerdctl.go file:
`
func resolveIP(fc *config.Finch, host string, logger flog.Logger) string {
	parts := strings.SplitN(host, ":", 2)
	// If the IP Address is a string called "host-gateway", replace this value with the IP address that can be used to
	// access host from the containers.
	// TODO: make the host gateway ip configurable.
	if parts[1] == dockerops.HostGatewayName {
		resolveIP := *fc.HostGatewayIP
		logger.Debugf(`Resolving special IP "host-gateway" to %q for host %q`, resolveIP, parts[0])
		return fmt.Sprintf("%s%s", parts[0], resolveIP)
	}
	return host
}
`

Since I already change the config file to add that option in the Finch struct